scripts/generate_BERTMap_predictions.py

```diff
@@ -0,0 +1,495 @@
+"""Generate mappings using BERTmap."""
+
+import argparse
+import os
+from functools import reduce
+from itertools import product
+from pathlib import Path
+
+import click
+import torch
+from deeponto.align.bertmap import DEFAULT_CONFIG_FILE, BERTMapPipeline
+from deeponto.onto import Ontology
+from huggingface_hub import snapshot_download
+from pystow.utils import download
+from tqdm import tqdm
+
+import biomappings
+from biomappings.bertmap import (
+    PREFIX_TO_DOWNLOAD_URL,
+    SOURCE_PREFIX_IRI_MAPS,
+    get_luid,
+)
+from biomappings.resources import append_prediction_tuples
+
+HERE = Path(__file__).parent.resolve()
+ROOT = HERE.parent.resolve()
+BERTMAP_DIR = ROOT.joinpath("bertmap")
+BERTMAP_DIR.mkdir(exist_ok=True)
+
+ONTOLOGIES_DIRECTORY = BERTMAP_DIR.joinpath("resources")
+ONTOLOGIES_DIRECTORY.mkdir(exist_ok=True)
+
+
+# pull
+def download_ontologies(
+    target_ontology_train: str,
+    source_ontology_train: str,
+    source_ontologies_inference: list[str],
+    target_ontologies_inference: list[str],
+) -> dict[str, Path]:
+    """Download OWL Files for specified ontologies."""
+    prefix_to_path = {}
+    for prefix in tqdm(
+        [
+            target_ontology_train,
+            source_ontology_train,
+            *source_ontologies_inference,
+            *target_ontologies_inference,
+        ],
+        desc="Downloading ontologies",
+        leave=False,
+    ):
+        ext = ".ttl" if prefix.upper() == "MESH2024" else ".owl"
+        ontology_path = ONTOLOGIES_DIRECTORY.joinpath(prefix.lower()).with_suffix(ext)
+        prefix_to_path[prefix.lower()] = ontology_path
+        if not ontology_path.is_file():
+            tqdm.write(f"Downloading {prefix}")
+            download(url=PREFIX_TO_DOWNLOAD_URL[prefix.upper()], path=ontology_path)
+    return prefix_to_path
+
+
+def load_bertmap(
+    config_path: Path,
+    target_ontology_train: str,
+    source_ontology_train: str,
+    ontology_paths: dict[str, Path],
+    use_biomappings: bool = False,
+    train_model: bool = False,
+) -> BERTMapPipeline:
+    """Load in the bertmap model (will download from hugging face if not present in ./bertmap)."""
+    config = BERTMapPipeline.load_bertmap_config(config_path.as_posix())
+
+    if use_biomappings:
+        print("-" * 100)
+        print("generating known mappings")
+        config.known_mappings = save_known_maps(
+            target_ontology_train=target_ontology_train,
+            source_ontology_train=source_ontology_train,
+            mappings_directory=BERTMAP_DIR.joinpath("knownMaps"),
+        ).as_posix()
+    print("-" * 100)
+    print(f"Using config: \n{config}")
+    print("-" * 100)
+    if not train_model:
+        config.global_matching.enabled = False
+        if not os.path.isdir("bertmap"):
+            print("downloading model from hugging face")
+            snapshot_download(repo_id="buzgalbraith/BERTMAP-BioMappings", local_dir="./")
```

Comment:
where does this come from? is it the result of some of the code that's run here? if so, how was it uploaded to huggingface?